components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/ClaimMetadataManagementServiceImpl.java

```diff
@@ -648,6 +652,96 @@ public List<Claim> getMappedExternalClaimsForLocalClaim(String localClaimURI, St
+           }
+        }
+
+        String[] allowedProfilePropertyKeys = {
+                ClaimConstants.SUPPORTED_BY_DEFAULT_PROPERTY,
+                ClaimConstants.REQUIRED_PROPERTY,
+                ClaimConstants.READ_ONLY_PROPERTY
+        };
+        for (String propertyKey: allowedProfilePropertyKeys) {
+            syncAttributeProfileProperties(claimProperties, allowedClaimProfiles, propertyKey);
+        }
+    }
+
+    /**
+     * Remove the profile properties if the profile property value is the same as the global property value.
+     * If all the profile properties have the same value, change the global property value to the profile value.
+     *
+     * @param claimProperties      Map of claim properties to be updated.
+     * @param allowedClaimProfiles Set of allowed claim profiles.
+     * @param propertyKey          Property key to be updated.
+     */
+    private void syncAttributeProfileProperties(Map<String, String> claimProperties, Set<String> allowedClaimProfiles,
+                                                String propertyKey) {
+
+        String globalValue = claimProperties.get(propertyKey);
+        boolean isAllProfilesHaveSameValue = true;
+        boolean isAtLeastOneProfileValueMatchingGlobal = false;
+        String commonProfileValue = null;
+
+        for (String profileName : allowedClaimProfiles) {
+            String profilePropertyKey = ClaimConstants.PROFILES_CLAIM_PROPERTY_PREFIX + profileName +
+                    ClaimConstants.CLAIM_PROFILE_PROPERTY_DELIMITER + propertyKey;
+            String profilePropertyValue = claimProperties.get(profilePropertyKey);
+
+            // Remove the profile property if it is the same as the global property.
+            if (StringUtils.equals(globalValue, profilePropertyValue)) {
+                claimProperties.remove(profilePropertyKey);
+                isAtLeastOneProfileValueMatchingGlobal = true;
+                continue;
+            }
+
+            // If we've already found at least one profile that matches the global, skip further consistency check.
+            if (isAtLeastOneProfileValueMatchingGlobal) {
+                continue;
+            }
+
+            if (StringUtils.isBlank(profilePropertyValue)) {
+                isAllProfilesHaveSameValue = false;
+                continue;
+            }
+            if (commonProfileValue == null) {
+                commonProfileValue = profilePropertyValue;
+            } else if (!StringUtils.equals(commonProfileValue, profilePropertyValue)) {
+                isAllProfilesHaveSameValue = false;
+            }
+        }
+
+        // If we never found a profile value that matches the global value, and all profiles share a common non-blank
+        // value, then overwrite the global property with the common value and remove the profile-specific properties.
+        if (!isAtLeastOneProfileValueMatchingGlobal && isAllProfilesHaveSameValue) {
```

Comment:
if (isAllProfilesHaveSameValue && !isAtLeastOneProfileValueMatchingGlobal) {
    
     // All the profiles have same value and the global value is different value. Hence update the global value.
     claimProperties.put(propertyKey, commonProfileValue);

    //  Remove the profile-specific properties as the global value denotes all the profile properties.
        allowedClaimProfiles.forEach(profile -> {
                String key = ClaimConstants.PROFILES_CLAIM_PROPERTY_PREFIX + profile +
                        ClaimConstants.CLAIM_PROFILE_PROPERTY_DELIMITER + propertyKey;
                claimProperties.remove(key);
            });

}

Reply:
Updated